src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -34,6 +49,31 @@ file_exists(const std::string& fnm)
 }
 
 namespace xclhwemhal2 {
+    //Thread for which pooling for transaction from SIM_QDMA
+    void hostMemAccessThread(xclhwemhal2::HwEmShim* inst);
+
+    //helper class for transactions from SIM_QDMA to XRT
+    class Q2H_helper {
+        private:
+        call_packet_info *ci_msg;
+        response_packet_info *ri_msg;
+	    size_t  i_len;
+	    void*   i_buf;
+	    size_t  ri_len;
+	    void*   ri_buf;
+	    size_t  buf_size;
+	    void*   buf;
```

Comment:
This looks like very unsafe handling of buffers.

Reply:
Now unique_ptr is used to manage the buffers